types/modules/currency-db.d.ts

```diff
@@ -0,0 +1,60 @@
+import { FirebotUser } from "./user-db";
+
+type Currency = {
+    id: string;
+    name: string;
+    active: boolean;
+    limit: number;
+    transfer: "Allow" | "Disallow";
+    interval: number;
+    payout: number;
+    /** Offline payout */
+    offline: number;
+    bonus: Record<string, number>;
+  };
+
+export type CurrencyDB = {
+    adjustCurrencyForUser: (
+        username: string, 
+        currencyId: string,
+        value: number,
+        adjustType?: string
+    ) => boolean,
+    addCurrencyToOnlineUsers: (
+        currencyId: string,
+        value: string,
+        ignoreDisposable?: boolean,
+        adjustType?: string
+    ) => Promise<void>,
+    getUserCurrencyAmount: (
+        username: string,
+        currencyId: string
+    ) => number,
+    purgeCurrencyById: (
+        currencyId: string
+    ) => void,
+    addCurrencyToNewUser: (
+        user: FirebotUser
+    ) => FirebotUser,
```

Comment:
```suggestion
```
I’m not sure if this function should be exported. Firebot calls it when a new user is created anyway, so there shouldn’t be a need to use this from a custom script.

Reply:
Cool, will take it out.  I was mostly trying to stay 1-1 with what I saw in Firebot.